crates/cli/cli/src/chainspec/mod.rs

```diff
@@ -2,6 +2,14 @@ use clap::builder::TypedValueParser;
 use reth_chainspec::ChainSpec;
 use std::sync::Arc;
 
+/// Ethereum chain specification parser.
+#[cfg(not(feature = "optimism"))]
+pub mod eth;
+
+/// Optimism chain specification parser.
+#[cfg(feature = "optimism")]
+pub mod optimism;
```

Comment:
we should move this to an optimism crate, I think, optimism-cli for now, because we need to extract this anyway. I'm fine with having duplicated code for now